src/vs/workbench/browser/parts/editor/tabFocus.ts

```diff
@@ -0,0 +1,71 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Emitter } from 'vs/base/common/event';
+import { Disposable } from 'vs/base/common/lifecycle';
+import { TabFocusContext, TabFocus } from 'vs/editor/browser/config/tabFocus';
+import { IConfigurationService } from 'vs/platform/configuration/common/configuration';
+import { RawContextKey, IContextKey, IContextKeyService } from 'vs/platform/contextkey/common/contextkey';
+import { TerminalSettingId } from 'vs/platform/terminal/common/terminal';
+
+export const terminalTabFocusContextKey = new RawContextKey<boolean>('terminalTabFocusMode', false, true);
+export const editorTabFocusContextKey = new RawContextKey<boolean>('editorTabFocusMode', false, true);
```

Comment:
Not a fan of having the terminal stuff tracked inside editor, can we move that to `workbench/contrib/terminal/`? Initially I thought this was going to cause layer breaker compile errors due to `TerminalSettingsId` being used, but I remembered we moved it to platform. Even still, I think it will be good for the codebase to keep all references to terminal out of `editor/`. If needed we can require the use of `terminalFocus`/`editorTextFocus` with the new context keys.

Reply:
Not sure what the right place is for this because we need to track both editor and terminal in this class.

Reply:
looking

Reply:
this class is used by `EditorStatus`, so it will not be possible to move it to `terminal`.

Reply:
I can move that context key though